homeassistant/components/sma/sensor.py

```diff
@@ -60,59 +829,43 @@ def __init__(
         self,
         coordinator: DataUpdateCoordinator,
         config_entry_unique_id: str,
+        description: SensorEntityDescription | None,
         device_info: DeviceInfo,
         pysma_sensor: pysma.sensor.Sensor,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
+        if description is not None:
+            self.entity_description = description
+        else:
+            self._attr_name = pysma_sensor.name
+
         self._sensor = pysma_sensor
-        self._enabled_default = self._sensor.enabled
-        self._config_entry_unique_id = config_entry_unique_id
-        self._attr_device_info = device_info
 
-        if self.native_unit_of_measurement == UnitOfEnergy.KILO_WATT_HOUR:
-            self._attr_state_class = SensorStateClass.TOTAL_INCREASING
-            self._attr_device_class = SensorDeviceClass.ENERGY
-        if self.native_unit_of_measurement == UnitOfPower.WATT:
-            self._attr_state_class = SensorStateClass.MEASUREMENT
-            self._attr_device_class = SensorDeviceClass.POWER
+        self._attr_device_info = device_info
+        self._attr_unique_id = (
+            f"{config_entry_unique_id}-{pysma_sensor.key}_{pysma_sensor.key_idx}"
+        )
 
         # Set sensor enabled to False.
         # Will be enabled by async_added_to_hass if actually used.
         self._sensor.enabled = False
 
     @property
     def name(self) -> str:
-        """Return the name of the sensor."""
+        """Return the name of the sensor prefixed with the device name."""
         if self._attr_device_info is None or not (
             name_prefix := self._attr_device_info.get("name")
         ):
             name_prefix = "SMA"
 
-        return f"{name_prefix} {self._sensor.name}"
+        return f"{name_prefix} {super().name}"
```

Comment:
In a future pr this can switch to use https://developers.home-assistant.io/docs/core/entity?_highlight=has_entity_name#has_entity_name-true-mandatory-for-new-integrations